### PR TITLE
Accept an invitation to an impact group (for existing users KIDS-724, temp users KIDS-725, and new users KIDS-722)

### PR DESCRIPTION
**Pull request by Daniela510**
## Description

- Impact groups are only fetched for US users
- New & Temp users are redirected to registration
- New & Temp users that are invited do not get the alert dialog, just the invitation bottomsheet
- Existing users are redirected to a success screen that then redirected to the family screen

Important: Api data is mocked

### CODE REVIEW

File: lib/core/enums/amplitude_events.dart [hunk 1]
```diff
@@ -2,6 +2,7 @@ enum AmplitudeEvents {
   letsGo('lets_go'),
   login('login'),
   familyClicked('family_clicked'),
+  clickedSeeMyFamily('clicked_see_my_family'),
```
Comment (pavelpavelko) [inline lib/core/enums/amplitude_events.dart:5]: perhaps would be more consistant seeMyFamilyClicked?)

File: lib/core/network/api_service.dart [hunk 2]
```diff
@@ -937,6 +937,65 @@ class APIService {
...
+    // final decodedBody = jsonDecode(response.body) as Map<String, dynamic>;
+    // final itemMap = decodedBody['items']! as List<dynamic>;
+    // return itemMap;
+    final mockData = jsonEncode({
+      "items": [
+        {
+          'id': "GUIDGUIDGUID",
+          "status": "Invited",
+          "name": "Luza Family",
+          "type": "Family",
```
Comment (pavelpavelko) [inline lib/core/network/api_service.dart:990]: shall we create enum for type?

File: lib/features/impact_groups/cubit/impact_groups_cubit.dart [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+          await _impactGroupInviteRepository.fetchImpactGroups();
+
+      emit(
+        state.copyWith(
+          status: ImpactGroupCubitStatus.fetched,
+          impactGroups: impactGroups,
+        ),
+      );
+    } catch (e) {
+      emit(
```
Comment (pavelpavelko) [inline lib/features/impact_groups/cubit/impact_groups_cubit.dart:38]: loggit?

File: lib/features/impact_groups/cubit/impact_groups_cubit.dart [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+          error: e.toString(),
+        ),
+      );
+    }
+  }
+
+  void checkForInvites() {
+    for (var i = 0; i < state.impactGroups.length; i++) {
+      if (state.impactGroups[i].status == ImpactGroupStatus.invited) {
+        emit(
```
Comment (pavelpavelko) [inline lib/features/impact_groups/cubit/impact_groups_cubit.dart:50]: just wondering how the logic works for multiple invitation at the same time?

File: lib/features/impact_groups/cubit/impact_groups_cubit.dart [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    required String groupId,
+  }) async {
+    emit(state.copyWith(status: ImpactGroupCubitStatus.loading));
+    try {
+      await _impactGroupInviteRepository.acceptGroupInvite(
+        groupId: groupId,
+      );
+      await fetchImpactGroups();
+    } catch (e) {
+      emit(
```
Comment (pavelpavelko) [inline lib/features/impact_groups/cubit/impact_groups_cubit.dart:71]: loggit

File: lib/features/impact_groups/impact_group_join_success_page.dart [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+                        );
+                      }),
+                )
+              else
+                Column(
+                  children: [
+                    const DownloadG4KButton(),
+                    TextButton(
+                      onPressed: () {
+                        context.pushReplacementNamed(
```
Comment (pavelpavelko) [inline lib/features/impact_groups/impact_group_join_success_page.dart:123]: shall we track amplitude here as well?

File: lib/features/impact_groups/impact_group_recieve_invite_sheet.dart [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+                        queryParameters: {
+                          'email': user.email,
+                          'createStripe': createStripe.toString(),
+                        },
+                      )
+                      ..pop();
+                  } else {
+                    context
+                      ..goNamed(Pages.joinImpactGroupSuccess.name)
+                      ..pop();
```
Comment (pavelpavelko) [inline lib/features/impact_groups/impact_group_recieve_invite_sheet.dart:76]: I do not understand why do we need this pop() right after goNamed?

Comment (Daniela510) [reply lib/features/impact_groups/impact_group_recieve_invite_sheet.dart:76]: It closes the bottomsheet after navigating. If you remove the pop then swipe back you can see the bottomsheet open over the homescreen.

File: lib/features/impact_groups/cubit/impact_groups_cubit.dart (hunk 4 above)
Comment (Daniela510) [reply lib/features/impact_groups/cubit/impact_groups_cubit.dart:50]: Only one bottomsheet will pop up, the next one will pop up the next time they go to home page or they logout/login

**Pull request merged**